package/network/services/dnsmasq/files/dnsmasq.init

```diff
@@ -461,16 +461,18 @@ dhcp_tag_add() {
 	# NOTE: dnsmasq has explicit "option6:" prefix for DHCPv6 so no collisions
 	local cfg="$1"
 
-	tag="$cfg"
-
-	[ -n "$tag" ] || return 0
+	config_get match_tag "$cfg" match_tag
+	[ -n "$match_tag" ] || match_tag="$cfg"
+	# if match_tag is empty in anonymous tag entry, bail:
+	[ -z "$match_tag" ] && [ -z "$cfg" ] && return 0
```

Comment:
Did you mean line 464?

Reply:
No

Reply:
Then the first test (`[ -n "$match_tag" ] ||`) on 465 is not redundant. Otherwise how would it help to assign match_tag on 464, then assign it on 465? The assign on 465 is conditional to the first test on 465. 

At least this code is tested and performs as anticipated.

Reply:
Okay, let's walk through the code.  First, `$cfg` will never be empty, or line 464 won't work.

So by the time the 2nd half of line 465 happens (the assignment), `$match_tag` is not empty.

So line 467 is superfluous because the first and 2nd tests will never be true.

Reply:
So the assumption is `$cfg` will never be empty. That was the whole problem: it sometimes *is* empty when the tag is anonymous, thus this PR.

Reply:
Then we shouldn’t call `config_get` if `$cfg` is empty…

Reply:
Sorry, got my wires crossed. Yeah, $cfg is never empty. 🤦